plat/mediatek/common/drivers/uart/8250_console.S

```diff
@@ -25,7 +34,7 @@
 	 * Clobber list : x1, x2, x3
 	 * -----------------------------------------------
 	 */
-func console_core_init
+func console_8250_core_init
```

Comment:
I don't think it's appropriate to use 16550_console.S. In addition to the difference in highspeed settings, there are other differences: like FCR registers, and the interface to log is not the same.

Reply:
Is there a significant difference in FCR? You're just clearing the FIFOs (which is something we could do on all boards, although it also shouldn't be necessary), and the 16550 driver is setting a DMA bit which I think doesn't make a difference either way. Not sure what you mean by "the interface to log"?

I think it's valuable to avoid code duplication wherever possible, so if possible implementing this driver as a thin wrapper (or a compile-time flag option) to the 16550 driver would be preferable.

Reply:
Have switched to 16550 uart driver.

Reply:
@jwerner-chromium and @kennyliang-mtk @mtk-rex-bc-chen 
We've found 16550 drivers seems cannot do baud rate at 921600, however, 8250 can do 921600. We've got many request hope MediaTek could support 921600 in TF-A. If there is no any concern, MediaTek will submit patch for switching driver back to 8250. At least, we hope to add a compile option in Makefile to enable 8250 and disable 16550.
Could you provide any comment? Thanks.

Reply:
Sorry, I'm not sure I really understand. We're talking about two driver implementations that would each talk to the same hardware peripheral, right? I see no fundamental reason why your implementation should be able to support a baud rate that the generic 16550 driver cannot.

Please identify what exactly your driver implementation is doing differently from the generic 16550 driver that allows it to support this baud rate. Once we know that, we can hopefully find a way to fix the generic driver without requiring diverging implementations.

Reply:
@macpaul-lin-mtk The real 16550 is actually a successor of 8250 IC, so I'd assume your '8250' (driver) is 'generic 8250-like driver that supports MTK extensions'. Like Julius said, in that case you should extend the generic UART driver (although it was named as 16550, it is actually 8250/16550 shared driver) so it can support the highspeed baudrate.

Reply:
@hungte @jwerner-chromium 
Thanks for the clarification, I'll inform our colleague to extend 'generic 8250-like driver that supports MTK extensions' on 16550. I've found there might be some register conflict on offset 0x24 ((HIGHSEEED/FIFO) but I'm not familiar with this. Let's see review the patch from MediaTek later for 16550. Thanks .